feat: implement edly login/registration flow

Waleed-Mujahid opened the pull request:
## Description
- use `EdlyLogistrationInfo` to show meaningful errors during logistration flow
- use `EmailCheckWidget` to allow user to enter email before login/register page
- Re-route user to login / register based on backend api response

## Ticket
https://projects.arbisoft.com/project/edly-product/task/8082
https://projects.arbisoft.com/project/edly-product/task/8089


## Linked PR
https://github.com/edly-io/edly-features-app/pull/39
https://github.com/edly-io/frontend-saas-widgets/pull/3

Waleed-Mujahid committed:
feat: implement edly login/registration flow

- Added Edly reducer and saga to the root reducer and saga respectively.
- Updated LoginPage to utilize Edly's prefilled email feature and display EdlyAlert.
- Enhanced Logistration component to conditionally render EmailCheckPage based on props.
- Incorporated EdlyAlert in RegistrationPage for consistent user feedback.

Anas-hameed commented on src/edly/data/reducers.js:
```diff
@@ -0,0 +1,61 @@
+  prefilledEmail: '',
+  errorCode: null,
+  context: {},
+};
```
We should add a sub-level that identifies the specific component this reducer belongs to. The current edly namespace doesn’t indicate which component uses this code.

`src/edly/data/reducers.js` is not indicating the component hierarchy

Waleed-Mujahid replied:
I have moved the code into a `EmailCheckWidget` widget

Anas-hameed reviewed (changes requested):
We should have one more level

Waleed-Mujahid committed:
feat: refactor Edly logistration logic

- auto populate email for new user in registration form
- auto populate email for existing user in login form
- show EmailCheckWidget even if public reg is disabled

muhammadali286 commented on src/data/reducers.js:
```diff
@@ -1,3 +1,4 @@
+import { emailCheckReducer, emailCheckStoreName } from '@anas_hameed/edly-saas-widget';
```
Add TODO for the change of import. When we shift it to edly-io package.

Waleed-Mujahid replied:
done

Waleed-Mujahid committed:
chore: add TODO comments for import updates

- Added TODO comments in multiple files to indicate the need for changing imports when the package is published to edly-io.

Waleed-Mujahid committed:
feat: enhance Logistration component with activation message support

- Added activation message handling by integrating getActivationStatus.
- Included AccountActivationMessage and ResetPasswordSuccess components in EmailCheckWidget for improved user feedback.

Waleed-Mujahid committed:
feat: Hide login tab when redirectTo is 'register'

This is done because when a user enters an email that does not exist on the current tenant, we want them to register and completly disable login.

If they want to login with a particular email they must come via the EdlyLogistration component that allows safe login and restricts cross tenant access
